### PR TITLE
Invitations private subspace

### PR DESCRIPTION
**Pull request by techsmyth**
Relates/Blocked by: https://github.com/alkem-io/server/pull/4614

To get the information needed about an invitation to a subspace from the returned field, not a direct lookup.

Remove usage of JourneyLevel, replaced with the actual SpaceLevel enum that is provided by the api. 

## Summary by CodeRabbit

- **New Features**
	- Introduced `SpaceLevel` enum for improved categorization of search results and community interactions.
	- Enhanced rendering logic to utilize `SpaceLevel` across various components, improving user experience in navigation and membership features.

- **Bug Fixes**
	- Resolved inconsistencies by replacing `journeyLevel` with `spaceLevel` in multiple interfaces and components.

- **Chores**
	- Updated types and imports to streamline code and enhance maintainability.

### CODE REVIEW

File: src/domain/community/application/applicationButton/ApplicationButton.tsx [hunk 1]
```diff
@@ -140,7 +141,7 @@ export const ApplicationButton = forwardRef<HTMLButtonElement | HTMLAnchorElemen
       extended
         ? t('components.application-button.extendedMessage', {
             join: verb,
-            journey: journeyLevel > 0 ? t('common.subspace') : t('common.community'),
+            journey: isSpace ? t('common.subspace') : t('common.community'),
```
Comment (bobbykolev) [inline src/domain/community/application/applicationButton/ApplicationButton.tsx:144]: condition was switched here, it should be the other way around (community for space and subspace translation for space)

**Review by bobbykolev — APPROVED** [review]
The invitations/applications were tested.

**Comment by valentinyanakiev** [comment]
@techsmyth I don't understand the need to regress to the old, obsolete space / challenge / opportunity. We've spent a ton of time to get away from it.

**Comment by techsmyth** [comment]
@valentinyanakiev we have not regressed. The numbers are the same. What I object to is having the numbers 0,1,2 hard coded around the code base. There is a separate issue to be picked up very near term to rename the enum entries - that is now a trivial change after this PR

**Review by bobbykolev — APPROVED** [review]
Re the discussion about the SpaceLeve, any further changes will be taken out as part of the Challange/Opportunity tech debt story.

**Pull request merged**